data_structures/skip_list.cpp

```diff
@@ -0,0 +1,207 @@
+/**
+ * A skip list is a data structure that is used for storing a sorted list of items with a
+ *  help of hierarchy of linked lists that connect increasingly sparse subsequences of the items
+ * 
+ *  References used: GeeksForGeeks skip list code, https://iq.opengenus.org/skip-list/ Pseudo Code.
+*/
+
+#include <iostream>
+#include <vector>
+#include <string.h>
+#include <stdlib.h>
```

Comment:
there are still some C headers in the code

Reply:
Yes, they are needed for memset function

Reply:
every C header has a corresponding C++ analog.
* for `string.h`, it will be `cstring` -> provides `std::memset`
* for `stdlib.h`, it will be `cstdlib` -> why this?

Reply:
Oh, I didn't know that.
Including stdlib.h is just  habit. It's not needed here.

Reply:
Memory issues shouldn't be the case